File: src/main/java/ru/stnk/RestTestAPI/service/UserDetailsServiceImpl.java
package ru.stnk.RestTestAPI.service;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;
import org.springframework.stereotype.Service;
import ru.stnk.RestTestAPI.entity.User;
import ru.stnk.RestTestAPI.repository.UserRepository;

import java.util.Optional;

@Service
public class UserDetailsServiceImpl implements UserDetailsService {

    @Autowired
    private UserRepository userRepository;

    @Override
    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {

        Optional userFromDB = userRepository.findByEmail(email);

        if (userFromDB.isPresent()) {

            return new UserDetailsImpl( (User) userFromDB.get() );

        } else {
            throw new UsernameNotFoundException(email);
        }
    }
}

File: src/main/kotlin/ru/stnk/resttestapi/service/UserDetailsServiceImpl.kt
package ru.stnk.resttestapi.service

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.security.core.userdetails.UserDetails
import org.springframework.security.core.userdetails.UserDetailsService
import org.springframework.security.core.userdetails.UsernameNotFoundException
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional
import ru.stnk.resttestapi.repository.UserRepository

@Service
class UserDetailsServiceImpl : UserDetailsService {

    @Autowired
    val userRepository: UserRepository? = null

    @Transactional
    @Throws(UsernameNotFoundException::class)
    override fun loadUserByUsername(email: String): UserDetails {

        val userFromDB = userRepository!!.findByEmail(email)

        if (userFromDB.isPresent) {

            return UserDetailsImpl.build(userFromDB.get())

        } else {
            throw UsernameNotFoundException(email)
        }
    }
}
